src/pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -591,15 +590,10 @@ def __resolve_ckpt_dir(self, trainer: "pl.Trainer") -> None:
         if trainer._weights_save_path_internal != trainer.default_root_dir:
             # the user has changed weights_save_path
             ckpt_path = os.path.join(trainer._weights_save_path_internal, "checkpoints")
-        elif trainer.loggers:
-            if len(trainer.loggers) == 1:
-                assert trainer.logger is not None
-                save_dir = trainer.logger.save_dir or trainer.default_root_dir
-            else:
-                save_dir = trainer.default_root_dir
-
-            name = _name(trainer.loggers)
-            version = _version(trainer.loggers)
+        elif len(trainer.loggers) > 0:
+            save_dir = trainer.loggers[0].save_dir or trainer.default_root_dir
```

Comment:
This implementation doesn't allow for `""` (empty string) as a valid `save_dir`. I think it should be (the local directory). This would have been a bug in the previous code too

Reply:
I make a new commit to make sure `save_dir` can be empty str.